nap/utils.py

```diff
@@ -60,19 +51,21 @@ def flatten_params(k, vs):
 
         # since we can have more than one value for a single key, we use a
         # tuple of two tuples instead of a dictionary
-        params_tuple = tuple(itertools.chain(*flat_params))
-        param_string = urllib.parse.urlencode(params_tuple)
+        params_tuple = tuple(sorted(itertools.chain(*flat_params), key=lambda key_value: key_value[0]))
```

Comment:
The previous concern was that a Python dictionary is used to pass the parameters and it does not order its entries, but it also can't have more than one value for a key.

Maybe, for the sake of this PR we should stick to what we are already doing. If there is a good case for changing the behavior that can be done in a separate PR along with some real world data demonstrating why one approach is better or worse than the other.